src/module/combat/combat.js

```diff
@@ -1192,11 +775,11 @@ Hooks.on('renderCombatTracker', (app, html, data) => {
             ev.preventDefault();
             console.log("Starfinder | Rendering Encounter Statistics Dialog");
             const isStarship = activeCombat.flags.sfrpg.combatType === "starship";
-            const contentTemplate = `//systems/sfrpg/templates/combat/${isStarship ? "starship" : "normal"}-encounter-stats.hbs`;
+            const contentTemplate = `//systems/sfrpg/templates/apps/${isStarship ? "starship" : "normal"}-encounter-stats.hbs`;
 
             new Dialog({
-                title: `${game.i18n.format("SFRPG.Combat.Difficulty.Tooltip.Details")}: ${CONFIG.SFRPG.difficultyLevels[activeCombat.flags.sfrpg.difficulty]} ${game.i18n.format("SFRPG.Combat.Difficulty.Tooltip.DifficultyEncounter")}`,
-                content: await renderTemplate(contentTemplate, activeCombat),
+                title: `${game.i18n.format("SFRPG.Combat.Difficulty.Tooltip.Details")}: ${CONFIG.SFRPG.difficultyLevels[diffObject.difficultyData.difficulty]} ${game.i18n.format("SFRPG.Combat.Difficulty.Tooltip.DifficultyEncounter")}`,
+                content: await renderTemplate(contentTemplate, diffObject),
```

Comment:
Nearly there, the point of the Application is to render it *with* the Application, not just use an instance of a class as the Dialog data. 
You'll want to instantiate a CombatDifficulty here and `.render(true)` it (or just render it if you need to instantiate it earlier). Then in CombatDifficulty, you'll want to create the data object used in the handlebars in the `getData` method. You can also create getters for the `title` and `template`

Reply:
In that case just instantiate earlier, then render it here. You can keep the methods to add the button display to the combat tracker on the CombatSFRPG class itself

Reply:
Fixed in
https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/1076/commits/8d9790b320d4d213c413d60bcec61feb11a4da2e